### ISSUE TITLE
userEvent.selectOptions causes RangeError: Maximum call stack size exceeded 

### ISSUE THREAD

**Issue by FLT4613** [testing-library/user-event#278]
There are two select box components with <Label>:

`<Label>`'s child element
```html
<label>
  Select
  <select>
    <option value="aaa">aaa</option>
    <option value="bbb">bbb</option>
  </select>
</label>
```
using `htmlFor`
```html
<>
  <label htmlFor="select-id">Select</label>
  <select id="select-id">
    <option value="aaa">aaa</option>
    <option value="bbb">bbb</option>
  </select>
</>
```

And  I test those with this code:
```js
const select = screen.getByLabelText("Select");
userEvent.selectOptions(select, "bbb");
expect(screen.getByText("bbb").selected).toBeTruthy();
```

But only `<Label>`'s child element test failed with error:
```
Maximum call stack size exceeded

RangeError: Maximum call stack size exceeded
    at innerInvokeEventListeners (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:1334515)
    at invokeEventListeners (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:1333311)
    at HTMLSelectElementImpl._dispatch (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:1332307)
    at fireAnEvent (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2293593)
    at sendClickToAssociatedNode (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2454970)
    at HTMLLabelElementImpl._activationBehavior (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2455952)
    at HTMLSelectElementImpl._dispatch (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:1332682)
    at fireAnEvent (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2293593)
    at sendClickToAssociatedNode (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2454970)
    at HTMLLabelElementImpl._activationBehavior (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2455952)
    at HTMLSelectElementImpl._dispatch (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:1332682)
    at fireAnEvent (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2293593)
    at sendClickToAssociatedNode (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2454970)
    at HTMLLabelElementImpl._activationBehavior (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2455952)
    at HTMLSelectElementImpl._dispatch (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:1332682)
    at fireAnEvent (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2293593)
    at sendClickToAssociatedNode (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2454970)
    at HTMLLabelElementImpl._activationBehavior (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2455952)
    at HTMLSelectElementImpl._dispatch (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:1332682)
    at fireAnEvent (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2293593)
    at sendClickToAssociatedNode (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2454970)
    at HTMLLabelElementImpl._activationBehavior (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2455952)
    at HTMLSelectElementImpl._dispatch (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:1332682)
    at fireAnEvent (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2293593)
    at sendClickToAssociatedNode (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2454970)
    at HTMLLabelElementImpl._activationBehavior (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2455952)
    at HTMLSelectElementImpl._dispatch (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:1332682)
    at fireAnEvent (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2293593)
    at sendClickToAssociatedNode (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2454970)
    at HTMLLabelElementImpl._activationBehavior (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2455952)
    at HTMLSelectElementImpl._dispatch (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:1332682)
    at fireAnEvent (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2293593)
    at sendClickToAssociatedNode (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2454970)
    at HTMLLabelElementImpl._activationBehavior (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2455952)
    at HTMLSelectElementImpl._dispatch (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:1332682)
    at fireAnEvent (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2293593)
    at sendClickToAssociatedNode (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2454970)
    at HTMLLabelElementImpl._activationBehavior (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2455952)
    at HTMLSelectElementImpl._dispatch (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:1332682)
    at fireAnEvent (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2293593)
    at sendClickToAssociatedNode (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2454970)
    at HTMLLabelElementImpl._activationBehavior (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2455952)
    at HTMLSelectElementImpl._dispatch (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:1332682)
    at fireAnEvent (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2293593)
    at sendClickToAssociatedNode (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2454970)
    at HTMLLabelElementImpl._activationBehavior (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2455952)
    at HTMLSelectElementImpl._dispatch (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:1332682)
    at fireAnEvent (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2293593)
    at sendClickToAssociatedNode (https://5qhf0.csb.app/static/js/jsdom-4.0.0.min.js:1:2454970)
    at HTMLLabelElementImpl._activationBehavior 
```

Why the test failed ?

My code sandbox(I added file `App.test.js`): 
https://codesandbox.io/s/boring-hypatia-5qhf0

Thanks!

**Comment by rbusquet** [comment]
looks like this was fixed in jsdom 16: https://github.com/jsdom/jsdom/pull/2921

**Comment by kentcdodds** [comment]
Thanks @rbusquet!

@FLT4613, could you upgrade the version of jsdom you're using and verify this does fix the problem?

**Comment by FLT4613** [comment]
Fixed it by upgrading Jest from 25.3.0 to 26.0.1. Thanks all !
